crates/bevy_time/src/time.rs

```diff
@@ -92,57 +121,189 @@ impl Time {
+            raw_delta
+        };
+
+        if self.last_update.is_some() {
+            self.delta = delta;
             self.delta_seconds = self.delta.as_secs_f32();
+            self.delta_seconds_f64 = self.delta.as_secs_f64();
+            self.raw_delta = raw_delta;
+            self.raw_delta_seconds = self.raw_delta.as_secs_f32();
+            self.raw_delta_seconds_f64 = self.raw_delta.as_secs_f64();
+        } else {
+            self.first_update = Some(instant);
         }
 
-        self.time_since_startup = instant - self.startup;
-        self.seconds_since_startup = self.time_since_startup.as_secs_f64();
+        self.elapsed += delta;
+        self.elapsed_seconds = self.elapsed.as_secs_f32();
+        self.elapsed_seconds_f64 = self.elapsed.as_secs_f64();
+        self.raw_elapsed += raw_delta;
+        self.raw_elapsed_seconds = self.raw_elapsed.as_secs_f32();
+        self.raw_elapsed_seconds_f64 = self.raw_elapsed.as_secs_f64();
         self.last_update = Some(instant);
     }
 
-    /// The delta between the current tick and last tick as a [`Duration`]
+    /// Returns the [`Instant`] the app was started.
+    #[inline]
+    pub fn startup(&self) -> Instant {
+        self.startup
+    }
+
+    /// Returns the [`Instant`] when [`update`](Self::update) was first called, if it exists.
+    #[inline]
+    pub fn first_update(&self) -> Option<Instant> {
+        self.first_update
+    }
+
+    /// Returns the [`Instant`] when [`update`](Self::update) was last called, if it exists.
+    #[inline]
+    pub fn last_update(&self) -> Option<Instant> {
+        self.last_update
+    }
+
+    /// Returns how much time has advanced since the last [`update`](Self::update), as a [`Duration`].
     #[inline]
     pub fn delta(&self) -> Duration {
         self.delta
     }
 
-    /// The delta between the current and last tick as [`f32`] seconds
+    /// Returns how much time has advanced since the last [`update`](Self::update), as [`f32`] seconds.
     #[inline]
     pub fn delta_seconds(&self) -> f32 {
         self.delta_seconds
     }
 
-    /// The delta between the current and last tick as [`f64`] seconds
+    /// Returns how much time has advanced since the last [`update`](Self::update), as [`f64`] seconds.
     #[inline]
     pub fn delta_seconds_f64(&self) -> f64 {
         self.delta_seconds_f64
     }
 
-    /// The time from startup to the last update in seconds
+    /// Returns how much time has advanced since [`startup`](Self::startup), as [`Duration`].
     #[inline]
-    pub fn seconds_since_startup(&self) -> f64 {
-        self.seconds_since_startup
+    pub fn elapsed(&self) -> Duration {
+        self.elapsed
     }
 
-    /// The [`Instant`] the app was started
+    /// Returns how much time has advanced since [`startup`](Self::startup), as [`f32`] seconds.
     #[inline]
-    pub fn startup(&self) -> Instant {
-        self.startup
+    pub fn elapsed_seconds(&self) -> f32 {
```

Comment:
On `Timer`/`Stopwatch` the methods is called `elapsed_secs`.
We should choose one or the other.

Reply:
I see. Yeah, `elapsed_secs` would better match the existing API. 

Can other reviewers chime in here?
We'd also have to change `delta_seconds` to `delta_secs`, so that's more API breakage.

Also, are we trying to follow [`std::time`](https://doc.rust-lang.org/std/time/struct.Duration.html#method.as_secs) as convention?

Reply:
I personally prefer using `seconds` but I do think there's value in following what the rust std does.